examples/src/main/java/org/bitcoinj/examples/ForwardingService.java

```diff
@@ -112,66 +135,59 @@ public void start() {
         // Download the blockchain and wait until it's done.
         kit.startAsync();
         kit.awaitRunning();
+        wallet = kit.wallet();
```

Comment:
This smells like a race to me. Peeking at the `WalletAppKit` code, wallet init happens in a separate thread.

Also, further down you're using the cached variable (I think) asynchronously, potentially introducing another race?

Generally I think there is no need to cache `wallet` at all. `WalletAppKit` is meant as a kit, shielding you from some internals. So you just save the kit, and use operations on it. Nothing wrong with `kit.wallet()` imho.

Reply:
> This smells like a race to me. Peeking at the `WalletAppKit` code, wallet init happens in a separate thread.
> 
> Also, further down you're using the cached variable (I think) asynchronously, potentially introducing another race?

The `vWallet` member (returned by `wallet()`) is initialized during startup which is by definition complete after `.awaitRunning()`. So there is no race condition.

> Generally I think there is no need to cache `wallet` at all. `WalletAppKit` is meant as a kit, shielding you from some internals. So you just save the kit, and use operations on it. Nothing wrong with `kit.wallet()` imho.

`WalletAppKit` is basically a factory object for the `Wallet` class (that also handles shutdown). It does not shield you from any internals as you have to use `.kit.wallet()` for everything wallet related.  `vWallet`/`.wallet()` is not `final` but is essentially a Kotlin  `lateinit` that is initialized during the `STARTING` state of the Guava service.

I would like to see some refactoring of `WalletAppKit` because (as we're seeing) it _is_ a little bit confusing.  One idea is to make it more clearly an asynchronous `Wallet` factory (perhaps the `Wallet` instance could be the return value for a `CompletableFuture`.) Another is to have it directly implement a `Wallet` interface (once we have one.)

But for purposes of this PR, we can keep it simple and just use `.kit.wallet()`.